### ISSUE TITLE
Hitting API usage limit -- Need a resolution

### ISSUE THREAD

**Issue by girishsince86** [raidendotai/cofounder#39]
I'm hitting daily API usage limit on both Anthropic/OpenAI. Is there a way to supply a new API with different account and resume the build? 
![image](https://github.com/user-attachments/assets/4e24ba37-7d13-425e-b54a-b5a232627d8a)

First attempt, i provided both Open AI and anthropic APIs.. only Anthropic APIs got used and build stopped once anthropic usage limit is hit. Can it switch to openAI once it hit limit on Anthropic? if so, how to do it?
Second attempt, provided OpenAI API only, and it hit the usage limit and build stopped again. Can i supply a nother OpenAI API from a different account and continue the same build? how to do it? 

Please help resolve this.! I'm eager to find out the quality of the build once the build is complete and i can run the application it built.

**Comment by BioCruz-sudo** [comment]
I used both anthropic and openai api with 20$ on each and my first run ended up having anthropic api be the only one used and eat up a lot of my usage. Capped at 1mil token a day then it switches to OpenAI API which maxed out quickly too, seems like the product, feature, and backend requirement documents are too many tokens for a single prompt

**Comment by radrad** [comment]
[I also reported this issue as well image limit of 5mb](https://github.com/raidendotai/cofounder/issues/40)
If you read my issue report, I suggested how this condition can be handled.

**Comment by feather812002** [comment]
> I used both anthropic and openai api with 20$ on each and my first run ended up having anthropic api be the only one used and eat up a lot of my usage. Capped at 1mil token a day then it switches to OpenAI API which maxed out quickly too, seems like the product, feature, and backend requirement documents are too many tokens for a single prompt

I got almost  trouble , I try to generate a test project, so after I input the project name and project summary, it starting working , but it fast hit the token limit and my API bill increase fast.

I suggestion let user can input PRD(product requirement document), then starting generate the project, if only a project name or a simple project summary describe, I sure the generate project not meet the requirement of product.

If generate project can meet the PRD 80%, I don't mind spend 100$,200$ to generate it , but obvious for Initial version it can't do this .

**Comment by radrad** [comment]
@feather812002 Good idea, but also we can start by getting insights into some business by only providing hight level description and we might learn a lot about what comes out. And then you can cherry pick what you want and possibly change PRD and ask the cofounder engine to start from updated PRD. So I would vote for such a feature.
It is not clear why many versions are output and what is the reason (maybe ask the engine to provide a reasonable explanation why a new version was introduced. Is it part of redesign request and when enough is enough.
It would be good on individual level of files we can lock it and not change or maybe lock some parts of the text or code and let the engine work around it.

**Comment by raidendotai** [comment]
i agree - should be more modular at inference and have more breakpoints
really boils down to how the DAGs (and attached functions) are set up
let's do that :)
